pkg/minio/knowledgebase.go

```diff
@@ -16,53 +19,60 @@ import (
-	// GetConvertedFilePathInKnowledgeBase returns the path of the converted file in MinIO.
-	GetConvertedFilePathInKnowledgeBase(kbUID, ConvertedFileUID, fileExt string) string
-	// GetChunkPathInKnowledgeBase returns the path of the chunk in MinIO.
-	GetChunkPathInKnowledgeBase(kbUID, chunkUID string) string
+	SaveTextChunks(_ context.Context, kbUID, fileUID uuid.UUID, chunks map[ChunkUIDType]ChunkContentType) (destinations map[string]string, _ error)
 	// DeleteKnowledgeBase deletes all files in the knowledge base.
-	DeleteKnowledgeBase(ctx context.Context, kbUID string) chan error
-	// DeleteAllConvertedFilesInKb deletes converted files in the knowledge base.
-	DeleteAllConvertedFilesInKb(ctx context.Context, kbUID string) chan error
-	// DeleteAllUploadedFilesInKb deletes uploaded files in the knowledge base.
-	DeleteAllUploadedFilesInKb(ctx context.Context, kbUID string) chan error
-	// DeleteAllChunksInKb deletes chunks in the knowledge base.
-	DeleteAllChunksInKb(ctx context.Context, kbUID string) chan error
+	DeleteKnowledgeBase(_ context.Context, kbUID uuid.UUID) chan error
+	// DeleteConvertedFileByFileUID deletes a converted file by file UID.
+	DeleteConvertedFileByFileUID(ctx context.Context, kbUID, fileUID uuid.UUID) error
+	// DeleteTextChunksByFileUID deletes all the chunks extracted from a file.
+	DeleteTextChunksByFileUID(ctx context.Context, kbUID, fileUID uuid.UUID) error
 }
 
-// prefix
-const uploadedFilePrefix = "/uploaded-file/"
-const convertedFilePrefix = "/converted-file/"
-const chunkPrefix = "/chunk/"
+const convertedFileDir = "converted-file"
+const chunkDir = "chunk"
+
+func convertedFileBasePath(kbUID, fileUID uuid.UUID) string {
+	return filepath.Join(kbUID.String(), fileUID.String(), convertedFileDir)
+}
+
+// SaveConvertedFile saves a converted file to MinIO with the appropriate MIME
+// type, returning the path of the saved file.
+func (m *Minio) SaveConvertedFile(ctx context.Context, kbUID, fileUID, convertedFileUID uuid.UUID, fileExt string, content []byte) (string, error) {
+	filename := convertedFileUID.String() + "." + fileExt
+	path := filepath.Join(convertedFileBasePath(kbUID, fileUID), filename)
 
-// SaveConvertedFile saves a converted file to MinIO with the appropriate MIME type.
-func (m *Minio) SaveConvertedFile(ctx context.Context, kbUID, convertedFileUID, fileExt string, content []byte) error {
-	filePathName := m.GetConvertedFilePathInKnowledgeBase(kbUID, convertedFileUID, fileExt)
 	mimeType := "application/octet-stream"
 	if fileExt == "md" {
 		mimeType = "text/markdown"
 	}
 
-	err := m.UploadBase64File(ctx, config.Config.Minio.BucketName, filePathName, base64.StdEncoding.EncodeToString(content), mimeType)
+	err := m.UploadBase64File(ctx, config.Config.Minio.BucketName, path, base64.StdEncoding.EncodeToString(content), mimeType)
 	if err != nil {
-		return err
+		return "", err
 	}
-	return nil
+
+	return path, nil
```

Comment:
🗒️ The path is returned here to avoid exporting `GetConvertedFilePathInKnowledgeBase`. The receiver (in the worker) will store the path as the `destination` column in the database, and that will be the source of truth for accessing the blob, not some code that may be updated in a backwards-incompatible way.